### PR TITLE
Add concourse pipeline files for gpcloud

### PR DESCRIPTION
**Pull request by adam8157**
Signed-off-by: Haozhou Wang <hawang@pivotal.io>
Signed-off-by: Yuan Zhao <yuzhao@pivotal.io>
Signed-off-by: Adam Lee <ali@pivotal.io>

### CODE REVIEW

File: concourse/tasks/regression_tests_gpcloud.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+platform: linux
+image_resource:
+  type: docker-image
+  source:
+    repository: pivotaldata/centos67-gpdb-gcc6-llvm-image
+    username: gpdb
+    password: a7vBWDg3mlMv
```
Comment (hlinnaka) [inline concourse/tasks/regression_tests_gpcloud.yml]: The password probably shouldn't be here. All the resources needed to build GPDB need to be public, so no password should be required in the first place.

**Comment by hlinnaka** [comment]
Please add comments explaining what this is all about.

After PR #1343 is pushed, this pipeline fill will need to be removed, and the gpcloud tests will need to hooked up to the top-level installcheck-world target instead. @schubert, can you pick up / arrange for someone else to pick up PR #1343 real soon, before we accumulate a ton more pipelines?

**Comment by adam8157** [comment]
Files in tasks and scripts could be reused by the main master pipeline. pipeline_gpcloud.yml is for anyone who wants to develop gpcloud and have an own pipeline.

**Comment by schubert** [comment]
@adam8157 & @yydzero "gpcloud" is a bit confusing of a name for something that only talks to S3. Is the intention to transform this into something that talks to multiple IaaS stores?

cc @ivannovick

**Comment by lij55** [comment]
Besides S3， we will need to support HDFS, Azure blob storage, GCP storage and other private s3 compatible storage service like Riak or swift. 
We change its name from s3ext.so because we are working on HDFS now.

**Comment by adam8157** [comment]
@hlinnaka We also have the need to run a standalone pipeline which only contains gpcloud tests, this PR doesn't conflict #1343 actually. What do you think?

**Comment by hlinnaka** [comment]
On 29 November 2016 04:14:16 EET, Adam Lee <notifications@github.com> wrote:
>@hlinnaka We also have the need to run a standalone pipeline which only
>contains gpcloud tests, this PR doesn't conflict #1343 actually. What
>do you think?

Why do you need a separate pipeline?

**Comment by adam8157** [comment]
A separate pipeline is useful for gpcloud agile development, since it's quite standalone. No worries, we will run installcheck-world also.

**Comment by Eulerizeit** [comment]
Does this mean we are going to be removing the Java dependencies in The GPDB MakeFiles as well as removing the testing overhead in the main pipeline?

**Comment by adam8157** [comment]
@Eulerizeit didn't get you. Nothing will be removed from the main pipeline.

This PR is because we also want a separate pipeline for gpcloud only, which could get the result much faster, good for the active gpcloud development. This could be done is also because gpcloud is quite standalone, doesn't impact or be impacted by other parts of GPDB most of the time.

**Comment by Eulerizeit** [comment]
Is gphdfs part of gpcloud? if so removing the tests for gphdfs from ICG would be nice. Right now in order to fully run the test suite on GPDB you have to hook into MVN and download Java etc. None of those things are fundamentally part of the database.

**Comment by lij55** [comment]
Hi @Eulerizeit , there would be two gphdfs in GPDB 5 before 'gphdfs2' is mature enough.
In early stage, gphdfs2 will lack of some features

**Comment by Eulerizeit** [comment]
I assume that means it is staying where it is till it is deprecated?

On Nov 29, 2016 10:10 PM, "Jasper Li" <notifications@github.com> wrote:

Hi @Eulerizeit <https://github.com/Eulerizeit> , there would be two gphdfs
in GPDB 5 before 'gphdfs2' is mature enough.
In early stage, gphdfs2 will lack of some features

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/greenplum-db/gpdb/pull/1383#issuecomment-263770914>,
or mute
the thread

**Comment by adam8157** [comment]
Yes, gphdfs will stay quite a while.

**Comment by adam8157** [comment]
merge because it doesn't conflict other PRs and we really need this.

**Pull request merged**